lib/matplotlib/axes/_axes.py

```diff
@@ -6535,6 +6529,39 @@ def hist(self, x, bins=None, range=None, density=None, weights=None,
 
         Other Parameters
         ----------------
+        color : color or array_like of colors or None, optional
+            The meaning of the parameter depends on *histtype*:
+
+            - For filled *histtype*\s ('bar', 'barstacked' and 'stepfilled'):
+
+              The facecolor of the bars. If providing multiple datasets to
+              *x*, there must be one color per dataset. If *None* the
+              standard line color sequence is used.
+
+              The *color* parameter is overridden if *facecolor* is given.
+
+            - For *histtype* 'step':
+
+              The linecolor of the step line. If None the standard line color
+              sequence is used.
+
+            Default: *None*
+
+        edgecolor : color or array_like of colors, optional
+            The edgecolor of the bars. If providing multiple datasets to
+            *x*, you can either pass a single edgecolor for all datasets or
+            one edgecolor per dataset. If not given, use :rc:`hist.edgecolor`.
+            If that is *None*, the default patch settings are used
+            (:rc:`patch.edgecolor` and :rc:`patch.force_edgecolor`).
+
+            Note: *edgecolor* is ignored for *histtype* 'step'. Use *color*
+            in this case.
+
+        linewidth : float, optional
+            The linewidth of the bar edges. If not given, use
+            :rc:`hist.linewidth`. If that is *None*, the default
+            :rc:`patch.linewidth` is used.
+
```

Comment:
I guess kwargs (just below) Patch properties if histtype = bar but Line2D properties if histtype = step?

I think the discussion above about color/edgecolor/facecolor is quite confusing.  I would suggest instead moving everything under the doc entry for `**kwargs` (which should read "Patch properties if histtype=bar; Line2D properties if histtype=step") and then have an additional paragraph like (haven't checked factual accuracy...)

> Color specification follows the underlying artist type.
> If using Patches (histtype=bar), the patch edge colors can be set using edgecolors (a single or list of colors), the patch face colors can be set using facecolors (...), and both can be simultaneously set using color (...); the defaults are ...
> If using Line2D (histtype=step), ...

(and whatever's right for stepfilled too...)